pkg/operator/config/config.go

```diff
@@ -0,0 +1,85 @@
+/*
+Copyright 2022.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package config
+
+import (
+	"os"
+	"path/filepath"
+	"time"
+
+	configv1 "github.com/openshift/api/config/v1"
+)
+
+const (
+	DefaultMetricsAddr          = ":8080"
+	DefaultEnableWebhook        = true
+	DefaultHealthProbeAddr      = ":8081"
+	DefaultRequeuePeriodSeconds = 5
+	DefaultEnableLeaderElection = false
+	// #nosec G101: Potential hardcoded credentials; path to token, not the content itself
+	DefaultTokenFile  = "/var/run/secrets/kubernetes.io/serviceaccount/token"
+	DefaultCACertFile = "/var/run/secrets/openshift.io/certs/service-ca.crt"
+)
+
+var DefaultCertDir = filepath.Join(os.TempDir(), "k8s-webhook-server", "serving-certs")
+
+// Config is configuration of the operator.
+type Config struct {
+
+	// MetricsBindAddress is the TCP address that the operator should bind to for
+	// serving prometheus metrics. It can be set to "0" to disable the metrics serving.
+	MetricsBindAddress string
+
+	// HealthProbeBindAddress is the TCP address that the operator should bind to for
+	// serving health probes (readiness and liveness).
+	HealthProbeBindAddress string
+
+	// The path of the service account token.
+	TokenFile string
+
+	// The path of the CA cert of the Agents' signing key pair.
+	CaCertFile string
+
+	// CertDir is the directory from where the operator loads keys and certificates.
+	CertDir string
```

Comment:
Where this config setting is used?

Reply:
I use it in main.go, instead of using multiple variables, I put them into the struct so that configurations should be clean and would be accessible for other packages as well.